packages/www/components/Dashboard/Logger/index.tsx

```diff
@@ -44,52 +62,83 @@ function orchestratorName({
   return !matches?.length ? address : matches[1];
 }
 
-function handleEvent(evt: events.Any, userIsAdmin: boolean): LogData[] {
-  switch (evt.type) {
-    case "stream_state":
-      const state = evt.state.active ? "active" : "inactive";
-      const region = evt.region || "unknown";
-      return [
-        {
-          key: evt.id,
-          timestamp: evt.timestamp,
-          level: "info",
-          text: `Stream is ${state} in region "${region}"`,
-        },
-      ];
-    case "transcode":
-      if (evt.success && !userIsAdmin) {
-        // non-admin users should only see fatal errors.
-        return [];
-      }
-      return evt.attempts
-        .filter((a) => a.error)
-        .map((a, idx) => ({
-          key: `${evt.id}-${idx}`,
-          timestamp: evt.timestamp,
-          level: "error",
-          text: `Transcode error from ${orchestratorName(a)} for segment ${
-            evt.segment.seqNo
-          }: ${a.error}`,
-        }));
-    case "webhook_event":
-      if (!evt.event.startsWith("multistream.")) {
-        console.error("unknown event:", evt.event);
-        break;
+function createEventHandler() {
+  const lastOrchestrator = useRef<string>();
+  const failedSegments = useRef(new Set<number>());
+
+  return useCallback(
+    function handleEvent(evt: events.Any, userIsAdmin: boolean): LogData[] {
+      switch (evt.type) {
+        case "stream_state":
+          const state = evt.state.active ? "active" : "inactive";
+          const region = evt.region || "unknown";
+          return [infoLog(evt, `Stream is ${state} in region "${region}"`)];
+        case "transcode":
+          const logs: LogData[] = [];
+          const seqNo = evt.segment.seqNo;
+
+          // non-admin users should only see fatal errors.
+          if (!evt.success || userIsAdmin) {
+            const errLogs = evt.attempts
+              .filter((a) => a.error)
+              .map((a, idx) => {
+                const orch = orchestratorName(a);
+                const msg = `Transcode error from ${orch} for segment ${seqNo}: ${a.error}`;
+                return errorLog(evt, msg, `error-${idx}`);
+              });
+            if (errLogs.length > 0) {
+              failedSegments.current.add(seqNo);
+              logs.push(...errLogs);
+            }
+          }
+
+          const lastAttempt = evt.attempts?.length
+            ? evt.attempts[evt.attempts.length - 1]
+            : null;
+          const orchestrator = orchestratorName(lastAttempt);
```

Comment:
This part is not clear to me, getting the orchestrator name passing the `lastAttempt` parameter to the `orchestratorName` function? What happens if `lastAttempt` is null? I am probably missing some context, so this is just a question

Reply:
So the `orchestratorName` function takes an attempt object, for easy usage here and especially on the `errLogs` transformations above, which were why they were originally created. If we want to be extra precise on the interface here, we should change it to receive only the `orchestrator` field of the attempt instead of the whole attempt, which is all that it needs. I haven't done it like this before cause the `errLogs` were very minimal, created on a single-line lambda, but now they're not so I'll update the interface for clarity :)

Also good catch on the `null` case, it was defeinitely not being handled and would explode here. It is kind of a overcaution here since we'll likely never have a null or empty attempts array anyway, but I'll keep it anyway. So I'm fixing those 2 things!